htdocs/product/class/html.formproduct.class.php

```diff
@@ -113,8 +118,10 @@ function loadWarehouses($fk_product=0, $batch = '', $status=null, $sumStock = tr
 			{
 				$obj = $this->db->fetch_object($resql);
 				if ($sumStock) $obj->stock = price2num($obj->stock,5);
+				$o = new Entrepot($this->db);
+				$o->fetch($obj->rowid);
 				$this->cache_warehouses[$obj->rowid]['id'] =$obj->rowid;
-				$this->cache_warehouses[$obj->rowid]['label']=$obj->label;
+				$this->cache_warehouses[$obj->rowid]['label']=$o->get_full_arbo();
```

Comment:
For a better compatibility and to match other dolibarr code part, i suggest 2 changes:
Keep here
$this->cache_warehouses[$obj->rowid]['label']=$obj->label;
and to introduce a new entry
$this->cache_warehouses[$obj->rowid]['parent_id']=$obj->fk_parent;

Then, instead of making a new and a fetch for each line, i suggest to make a function outside of loop to iniitalize the
$this->cache_warehouses[$obj->rowid]['fulllabel'] (label with full path) without doing any sql request at all (having a fetch for each line response is not possible for large database).
Solution is to loop on $this->cache_warehouses entries and for each of them calling a recursive function get_parent_path that search into $this->cache_warehouses for the entry that is the parent and call get_parent_path on it.